### PR TITLE
Add the ability to modify images in HSSF documents

### PR DESCRIPTION
**Pull request by MariusVolkhart**

### CODE REVIEW

**Review by shankera — APPROVED** [review]

File: poi/src/test/java/org/apache/poi/hssf/usermodel/TestHSSFPictureData.java [hunk 1]
```diff
@@ -17,23 +17,26 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
 import javax.imageio.ImageIO;
 
+import org.apache.poi.POIDataSamples;
 import org.apache.poi.POITestCase;
 import org.apache.poi.hssf.HSSFTestDataSamples;
+import org.apache.poi.hssf.record.crypto.Biff8EncryptionKey;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (PKJosh) [inline poi/src/test/java/org/apache/poi/hssf/usermodel/TestHSSFPictureData.java]: - [x] Fix wildcard import

File: poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java [hunk 2]
```diff
@@ -19,21 +19,152 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+        this.bStoreContainer = Objects.requireNonNull(bStoreContainer);
+    }
+
+    /**
+     * Initializes a {@link DrawingManager2} tied to the given workbook.
+     * <p>
+     * If the workbook does not yet have records for the Drawing Group, BStore, & DGG records, new records will be
+     * created and added to the workbook.
+     *
+     * @param workbook From which to initialize a the {@link DrawingManager2}.
```
Comment (MatthewFoster624) [inline poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java]:
```suggestion
     * @param workbook From which to initialize the {@link DrawingManager2}.
```

File: poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java [hunk 3]
```diff
@@ -19,21 +19,152 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+import org.apache.poi.ddf.EscherRecordTypes;
+import org.apache.poi.ddf.EscherSplitMenuColorsRecord;
+import org.apache.poi.hssf.record.CountryRecord;
+import org.apache.poi.hssf.record.DrawingGroupRecord;
+import org.apache.poi.hssf.usermodel.HSSFPictureData;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.util.Internal;
+import org.apache.poi.util.Removal;
+
+import static org.apache.poi.ss.usermodel.Workbook.*;
```
Comment (PKJosh) [inline poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java]: - [x] Fix wildcard import

**Comment by MariusVolkhart** [comment]
- [x] squash

**Review by PKJosh — APPROVED** [review]

File: poi/src/main/java/org/apache/poi/hssf/usermodel/HSSFWorkbook.java [hunk 4]
```diff
@@ -1958,80 +1943,13 @@ void initDrawings() {
...
-                break;
-            default:
-                EscherBitmapBlip blipRecordBitmap = new EscherBitmapBlip();
-                blipRecord = blipRecordBitmap;
-                blipRecordBitmap.setUID(uid);
-                blipRecordBitmap.setMarker((byte) 0xFF);
-                blipRecordBitmap.setPictureData(pictureData);
-                blipSize = pictureData.length + 25;
-                escherTag = (short) 0xFF;
-                break;
```
Comment (PKJosh) [inline poi/src/main/java/org/apache/poi/hssf/usermodel/HSSFWorkbook.java:1995]: Am I missing something about any of this being defaulted elsewhere? Or is no longer ~~unnecessary~~ necessary?

**Review by MatthewFoster624 — APPROVED** [review]

File: poi/src/main/java/org/apache/poi/hssf/usermodel/HSSFWorkbook.java (hunk 4 above)
Comment (MariusVolkhart) [reply poi/src/main/java/org/apache/poi/hssf/usermodel/HSSFWorkbook.java:1995]: Combination of defaulted elsewhere, previously redundant assignments, and no longer necessary, _I think_. We'll want to verify xls w/ images thoroughly in the product

**Pull request merged**